bibtex.html

```diff
@@ -0,0 +1,480 @@
+  author = {Daloze, Benoit and Marr, Stefan and Bonetta, Daniele and Mössenböck, Hanspeter},
+  booktitle = {Proceedings of the 2016 ACM International Conference on Object Oriented Programming Systems Languages \& Applications},
+  day = 2,
+  doi = {10.1145/2983990.2984001},
+  isbn = {978-1-4503-4444-9},
+  month = {November},
+  numpages = {18},
+  pages = {642--659},
+  url = {https://eregon.me/blog/assets/research/thread-safe-objects.pdf},
+  publisher = {ACM},
+  series = {OOPSLA'16},
+  title = {Efficient and Thread-Safe Objects for Dynamically-Typed Languages},
+  year = 2016,
+
+  vm:concepts = {VM Implementation Thread Safety},
+  vm:shortTitle = {Thread-Safe Object Model},
+  vm:edge:extends = {
+    Marr:15:DC for efficient sharing;
+    Woss:14:TOM + thread safety}
+}
+
+@article{Daloze:2018:TSC,
+  abstract = {Dynamic programming languages such as Python and Ruby are widely used, and much effort is spent on making them efficient. One substantial research effort in this direction is the enabling of parallel code execution. While there has been significant progress, making dynamic collections efficient, scalable, and thread-safe is an open issue. Typical programs in dynamic languages use few but versatile collection types. Such collections are an important ingredient of dynamic environments, but are difficult to make safe, efficient, and scalable.
+              In this paper, we propose an approach for efficient and concurrent collections by gradually increasing synchronization levels according to the dynamic needs of each collection instance. Collections reachable only by a single thread have no synchronization, arrays accessed in bounds have minimal synchronization, and for the general case, we adopt the Layout Lock paradigm and extend its design with a lightweight version that fits the setting of dynamic languages. We apply our approach to Ruby’s Array and Hash collections. Our experiments show that our approach has no overhead on single-threaded benchmarks, scales linearly for Array and Hash accesses, achieves the same scalability as Fortran and Java for classic parallel algorithms, and scales better than other Ruby implementations on Ruby workloads.},
+  author = {Daloze, Benoit and Tal, Arie and Marr, Stefan and Mössenböck, Hanspeter and Petrank, Erez},
+  doi = {10.1145/3276478},
+  journal = {Proceedings of the ACM on Programming Languages},
+  month = {November},
+  number = {OOPSLA},
+  pages = {108:1--108:30},
+  url = {https://eregon.me/blog/assets/research/thread-safe-collections.pdf},
+  series = {OOPSLA'18},
+  title = {{Parallelization of Dynamic Languages: Synchronizing Built-in Collections}},
+  volume = 2,
+  year = 2018,
+
+  vm:shortTitle = {Synchronizing Collections based on Reachability},
+  vm:edge:extends = {
+    Bolz:2013:SSC + thread safety;
+    Daloze:2016:TSO + collections},
+  vm:edge:uses = {
+    Daloze:2015:GLS + global synchronization;
+    Daloze:2016:TSO + tracking reachability}
```

Comment:
I think this edge should be an `extends` of the Domani reachability paper.
Something like Domani uses the concept `Tracking Reachability to Reduce Synchronization` and this paper extends Domani.
I didn't know about the Domani paper when writing this paper, but the technique is fairly similar so it could be seen as an extension of Domani. It also extends what's done with reachability, so I'd say `extends` over `uses` here.

`extends`/`uses` edges won't always be based on the author knowing about the previous paper beforehand, but I think that's fine.